R/DBConnection.R

```diff
@@ -106,6 +106,103 @@ setGeneric("dbSendQuery",
   valueClass = "DBIResult"
 )
 
+#' Execute a non-select query
+#'
+#' This function should be used when you want to execute a non-
+#' \code{SELECT} query on table (ex: \code{UPDATE}, \code{DELETE},
+#' \code{INSERT INTO}, \code{DROP TABLE}, ...). It will execute
+#' the query and return the number of rows affected by the operation.
+#'
+#' @inheritParams dbDisconnect
+#' @param statement a character vector of length 1 containing SQL.
+#' @return The number of rows affected by the \code{statement}
+#' @aliases dbExecQuery,DBIConnection,character-method
+#' @family connection methods
+#' @export
+#' @examples
+#' con <- dbConnect(RSQLite::SQLite(), ":memory:")
+#' dbWriteTable(con, "cars", head(cars, 3))
+#' dbReadTable(con, "cars")   # there's 3 rows!
+#' dbExecQuery(con, "INSERT INTO cars (speed, dist)
+#'                   VALUES (1, 1), (2, 2), (3, 3);")
+#' dbReadTable(con, "cars")   # there's now 6 rows!
+#' dbDisconnect(con)
+setGeneric("dbExecQuery",
+  def = function(conn, statement, ...) standardGeneric("dbExecQuery")
+)
+
+#' @export
+setMethod("dbExecQuery", signature("DBIConnection", "character"),
+  function(conn, statement, ...) {
+    if (grepl("select", tolower(statement))) {
```

Comment:
(I wonder what queries those would be, but granted people do get creative with code sometimes...)

If we don't validate the SQL at all, then that's that. But if we want to give a warning/print a message, why do you say that `dbGetRowsAffected()` can't be relied on? I had thought about using `dbHasCompleted()` first, but then came across [this](http://stackoverflow.com/questions/33735086/dbhascompleted-always-returns-true) -- is this a real problem? Or were these people doing something weird? From a very quick look I couldn't replicate the issue, but it did make me not want to use `dbHasCompleted()`.

So maybe by far the easiest thing might be not to do any check at all... A little uncomfortable with that, but I'll survive...